lib/gpu/lib/src/buffer.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:nativewrappers';
+import 'dart:typed_data';
+
+/// A reference to a byte range within a GPU-resident [Buffer].
+class BufferView {
+  /// The buffer of this view.
+  final HostBuffer buffer;
+
+  /// The start of the view, in bytes starting from the beginning of the
+  /// [buffer].
+  final int offsetInBytes;
+
+  /// The length of the view.
+  final int lengthInBytes;
+
+  /// Create a new view into a buffer on the GPU.
```

Comment:
its not really "on the gpu though"? For more functional pieces like the buffer view it may be easier to document once there are things that can be done with it. for example, "a view into a buffer that can be used for binding uniform data yada yada"

Reply:
I think `BufferView`s are effectively GPU-only because they'll only be used for binding buffers to `Command`s. I don't think we'd use a `BufferView` for updating a range on the host buffer, for example.